content/reference/dos-mitigation.md

```diff
@@ -0,0 +1,381 @@
+    - [Example screen recording of fail2ban in action](#example-screen-recording-of-fail2ban-in-action)
+    - [Setting Up fail2ban](#setting-up-fail2ban)
+  - [Leverage Resource Manager and a set of trusted peers to form an allow list (go-libp2p only)](#leverage-resource-manager-and-a-set-of-trusted-peers-to-form-an-allow-list-go-libp2p-only)
+- [Summary](#summary)
+
+# What we mean by a DOS attack
+
+A DOS attack is any attack that can cause your application to crash, stall, or
+otherwise fail to respond normally. An attack is considered viable if it takes
+fewer resources to execute than the damage it does. In other words, if the
+payoff is higher than the investment it is a viable attack and should be
+mitigated. Here are a few examples:
+
+1. A node opening many connections to a remote node and forcing that
+   node to spend 10x the compute time to handle the request relative to the
+   attacker node. This is attack viable because a single node amplifies its
+   affect 10x. This attack will continue to scale if the attacker adds more
+   nodes.
+
+2. 100 nodes asking a single node to do some work, but if this single node
+   goes down it will indirectly cause the loss of an asset. If the asset is more
+   valuable than the compute time of 100 nodes, this attack is viable.
+
+3. Many nodes connecting to a single node such that that node can no
+   longer accept new connections from an honest peer. This node is now
+   isolated from the honest peers in the network. This is commonly called an
+   eclipse attack and is viable if it's either cheap to eclipse this node, or if
+   eclipsing this node has a high payoff.
+
+Generally, the effect on our application can range from crashing to stalling to
+failing to handle new peers to degraded performance. Ideally we want
+our application to at worst suffer a slight performance penalty, but otherwise
+stay up and healthy.
+
+In the next section we'll cover some design strategies you should incorporate
+into your protocol to make sure your application stays up and healthy.
+
+# Incorporating DOS mitigation from the start
+
+The general strategy is to use the minimum amount of resources as possible and
+make sure that there's no untrusted amplification mechanism (e.g. an untrusted
+node can force you to do 10x the work it does). A protocol-level reputation
+system can help (take a look at [GossipSub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) for inspiration) as well as
+logging misbehaving nodes and actioning those logs separately (see fail2ban
+below).
+
+Here are some more specific recommendations
+
+## Limit the number of connections your application needs
+
+Each connection has a resource cost associated with it. A connection will
+usually represent a peer and a set of protocols with each their own resource
+usage. So limiting connections can have a leveraged effect on your resource
+usage.
+
+In go-libp2p the number of active connections is managed by the
```

Comment:
A few nitpics here:
1. We use connmgr and ConnManager
2. When we're hyperlinking, I think it's good to remove the ticks so it's clear that it's a hyperlink.  (see screenshot of rendering)
3. We do "ConnManager" no space and "Resource Manager" with space.
4. Maybe the first time we talk about Resource Manager here we make it a hyperlink?

<img width="827" alt="image" src="https://user-images.githubusercontent.com/85411/184450457-8eb69cb9-012f-4ed8-982d-d1f1bda8dbde.png">

Reply:
> We do "ConnManager" no space and "Resource Manager" with space.

In go-libp2p-core they're called `ConnManager` and `ResourceManager`. Using `Conn Manager` feels weird and so does `ResourceManager` although happy to make one change or the other if you feel strongly.

> When we're hyperlinking, I think it's good to remove the ticks so it's clear that it's a hyperlink. (see screenshot of rendering)

I think our template should support the fixed width + hyperlink. I'll see if I can fix it.

Reply:
> I think our template should support the fixed width + hyperlink. I'll see if I can fix it.

Fixed this by adding a `text-decoration: underline` css property.

<img width="786" alt="Screen Shot 2022-08-15 at 4 25 20 PM" src="https://user-images.githubusercontent.com/594035/184740948-2049b6e9-258d-4cdc-8735-58d54ba28f65.png">